### PR TITLE
Fixes #12112: Plugin for API authorization ACL

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/12112

### CODE REVIEW

File: api-authorizations/src/main/scala/bootstrap/rudder/plugin/ApiAuthorizationsConf.scala [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  // override default service level
+  RudderConfig.apiAuthorizationLevelService.overrideLevel(AclLevel)
+
+}
+
+/**
+ * Init module
+ */
+@Configuration
+class DataSourcesPluginConf extends Loggable with  ApplicationContextAware with InitializingBean {
```
Comment (VinceMacBuche) [inline api-authorizations/src/main/scala/bootstrap/rudder/plugin/ApiAuthorizationsConf.scala]: ApiAuthorizationsConf

File: api-authorizations/packaging/postinst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+# create table for Rudder data sources plugin:
+# note that if you installed Rudder 4.1~alpha or 4.1~beta1, or a previous version
+# of the plugin, you can skip that step (which would fail with a "table exists" error)
+psql -U rudder -h localhost -d rudder -f /opt/rudder/share/plugins/datasources/datasources-schema.sql
```
Comment (VinceMacBuche) [inline api-authorizations/packaging/postinst]: datasources

Comment (VinceMacBuche) [reply api-authorizations/packaging/postinst]: in fact we do not need this in postinst

File: api-authorizations/src/main/scala-templates/limited/com/normation/plugins/datasources/EnablePluginImpl.scala [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package com.normation.plugins.datasources
```
Comment (VinceMacBuche) [inline api-authorizations/src/main/scala-templates/limited/com/normation/plugins/datasources/EnablePluginImpl.scala]: package and directory are datasource

File: api-authorizations/src/main/scala/bootstrap/rudder/plugin/EnablePlugin.scala [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+package bootstrap.rudder.plugin
+
+import org.joda.time.DateTime
+
+/**
+ * This file defined an entry point for license information and other
+ * enabling of the plugin
+ */
+
+sealed trait DatasourcesStatus
```
Comment (VinceMacBuche) [inline api-authorizations/src/main/scala/bootstrap/rudder/plugin/EnablePlugin.scala]: Datasource all over in  the file!!

**Comment by fanf** [comment]
PR rebased

File: api-authorizations/src/main/scala-templates/limited/com/normation/plugins/datasources/EnablePluginImpl.scala [hunk 5]
```diff
@@ -0,0 +1,100 @@
+/*
```
Comment (VinceMacBuche) [inline api-authorizations/src/main/scala-templates/limited/com/normation/plugins/datasources/EnablePluginImpl.scala]: This file is in datasources directory!

**Comment by fanf** [comment]
PR rebased

**Pull request merged**